### ISSUE TITLE
Marshalling error of date in Scylla, works in Cassandra

### ISSUE THREAD

**Issue by tzach** [scylladb/scylla#4641]
Scylla version (or git commit hash): 3.0.8
OS (RHEL/CentOS/Ubuntu/AWS AMI): Docker

The following work with Cassandra 3.11 but not Scylla
```
cqlsh> CREATE KEYSPACE mykeyspace WITH replication = {'class': 'SimpleStrategy', 'replication_factor' : 1};
cqlsh> use mykeyspace ;
cqlsh:mykeyspace> CREATE TABLE test_date (a timestamp primary key); 
cqlsh:mykeyspace> INSERT INTO test_date (a) VALUES ('2019-07-02T18:50:10Z');
INSERT INTO test_date (a) VALUES ('2019-07-02T18:50:10Z'); 
InvalidRequest: Error from server: code=2200 [Invalid query] message="marshaling error: unable to parse date '2019-07-02T18:50:10Z': marshaling error: Unable to parse timezone 'z'"
```
Note the Z in the date is reported is in low case, altho the input is upper case
Cassandra does not accept low case either.

**Comment by tzach** [comment]
https://github.com/scylladb/scylla/blob/master/types.cc#L885

**Comment by slivne** [comment]
This was released in 3.0.9 and will be part of 3.1.0
